lib/vector-buffers/src/variants/disk_v2/tests/invariants.rs

```diff
@@ -41,8 +40,7 @@ async fn pending_read_returns_none_when_writer_closed_with_unflushed_write() {
                 .was_entered()
                 .finalize();
 
-            let mut blocked_read =
-                spawn(async move { reader.next().await.expect("read should not fail") });
+            let mut blocked_read = spawn(async move { read_next(&mut reader).await });
```

Comment:
Per a comment you left recently on a PR of mine, I think you could boil this, and other instances of this, down to `spawn(read_next(&mut reader))`?

Reply:
No, because the `reader` needs to be moved into the async block. If the function took `reader` by value then it could.